### ISSUE TITLE
Possible false-negative for `unused-argument` when function always raises

### ISSUE THREAD

**Issue by e-gebes** [pylint-dev/pylint#9935]
### Bug description

```python
def function1(value):  # argument is used
    return value


def function2(value):  # argument is unused, Pylint correctly reports it
    return 2


def function3(value):  # argument is unused, Pylint does not report
    raise ValueError
```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
************* Module a
5:14: W0613: Unused argument 'value' (unused-argument)
```

### Expected behavior

I would expect that also in case of an unconditional `raise` the argument would be reported as unused.

### Pylint version

```shell
pylint 2.17.7
astroid 2.15.8
Python 3.10.12 (main, Sep 11 2024, 15:47:36) [GCC 11.4.0]
```

**Comment by andreyfmc** [comment]
I think this will help you.
https://bit.ly/3TyLBMS
If you don't have the c compliator, install it.(gcc or clang)

**Comment by akamat10** [comment]
This can be reproduced with the latest pylint version. Seems to be a bug.
